### DISCUSSION TITLE
Receive device location info on the IoT device (Zephyr)

### DISCUSSION THREAD

**Discussion by rajeev1986** [1NCE-GmbH/1nce-iot-c-sdk#10] [Q&A]
I am an 1NCE customer and was curious how we could receive the device location retrieved by the cloud portal on the actual device itself that is running Zephyr RTOS? 
This would allows us to use 1nce service directly for fetching device location instead of using other 3rd party.
I could not find this sample in the 1nce sdk.

Any pointers or suggestions will be really great.
Thanks

**Comment by Mtalat** [comment]
Hi @rajeev1986 

We apologize for the delayed response. Unfortunately, due to an issue, we were not notified about your request on time.
 
To retrieve the position from the device:

- Obtain an access token from https://help.1nce.com/dev-hub/reference/postaccesstokenpost
- Get the device position from the API using the ICCID https://help.1nce.com/dev-hub/reference/get_v1-locate-devices-deviceid-positions

If you need any further assistance or a sample code, please let us know.

**Reply by rajeev1986** [reply]
Thanks @Mtalat, for the response. If you got or could share the sample code for the Zephyr, this would speedup my development and testing process.

**Reply by jamali005** [reply]
Hi @rajeev1986, 

We are currently working on the sample code for Zephyr and will get back to you as soon as it is ready. By the way, are you working with the nRF91 Dev kit?

**Reply by rajeev1986** [reply]
@jamali005, I am working with nRF9160, 9161, and also nRF9151 dev kits. So almost on all the cellular series of Nordic Semi.
In addition, we also have our [own devices](https://conexiotech.com/) that are based on all of the above and are connected by 1nce IoT SIMs.

Do let me know if you have a working sample. Our customers would love to get this new feature working. Thanks

**Comment by jamali005** [comment]
@rajeev1986 We’re working on the sample for the nRF9160 dev kit. We’ll update you as soon as we have it ready.
Thanks

**Comment by jamali005** [comment]
Hi @rajeev1986 ,
Good news, this feature is now available.
You can now retrieve the device’s last known location directly via the new 1NCE OS CoAP Location Endpoint
Documentation:
[1NCE OS CoAP Location Endpoint](https://help.1nce.com/docs/1nce-os/1nce-os-device-integrator/device-integrator-coap/?utm_source=chatgpt.com#location-endpoint)
I’ll close the discussion for now. If you need any further support, please feel free to reach out.
